src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -822,6 +822,16 @@ private void UpdateCallsiteExecutionState(Object callsiteData, RuntimeCore runti
             return compliantTarget;
         }
 
+        private Boolean IsCompatibleReplicationOption(List<ReplicationInstruction> oldOption, List<ReplicationInstruction> newOption)
+        {
+            if (oldOption.Count > 0 && newOption.Count > 0 && oldOption.Count < newOption.Count)
+            {
+                if (oldOption[0].ToString().Equals(newOption[0].ToString()))
```

Comment:
what does toString() evaluate to here? What does a repliationInstruction look like when stringified?

Reply:
ToString() returns the replicaion-option name and the number of indices like "Cartesian: indices:0" or "zipped:indices0,1".

Reply:
do this get called repeatedly in a tight loop? I would rather we avoid a string conversion if this is performance critical code - can we just override equals and compare the two properties?

Reply:
This wouldn't be called for every replication option. It will be called only when a replicationOption is already set in the previous trial and if it finds another match for the later replication trials. 
I can create a helper to compare those 2 properties but wouldn't be a lot different to the toStirng() method for this class. Let me know what you think.

Reply:
avoiding the toString is only important I think if this is called frequently 😄 - I don't know enough about this pathway to have a gut feeling about it - leave that up to you and some profiling / perf benchmarking.

Reply:
Yes. Will do good amount of performance testing before proceeding.

Reply:
this seems potentially scary - unless I am interpreting this wrong - 
https://github.com/DynamoDS/Dynamo/blob/ec10f936824152e7dd7d6d019efdcda0d78a5264/src/Engine/ProtoCore/Lang/Replication/ReplicationInstruction.cs#L36

it looks like if the zippedIndicies property were 1,000,000 items long, the toStrong() would be very slow.

Reply:
to follow up on this - calculating the toString() of a set of replication options seems very slow - 

I tried this
`replicationTrials.Select(x=>string.Join(",",x))`

over this:

```
x = [[[null,[1,2],[3,4]]]];
y = [[[[5,6],[7,8],[9,10]]]];
```

and passed it to `Point.ByCoordinates(double,double)`

producing this:
```
   [0]: ""
    [1]: "Cartesian, Index: 0"
    [2]: "Cartesian, Index: 0,Cartesian, Index: 0"
    [3]: "Cartesian, Index: 0,Cartesian, Index: 0,Cartesian, Index: 0"
    [4]: "Cartesian, Index: 0,Cartesian, Index: 0,Cartesian, Index: 0,Cartesian, Index: 0"
    [5]: "Cartesian, Index: 1"
    [6]: "Zipped, indecies: 0, 1 - Shortest"
    [7]: "Zipped, indecies: 0, 1 - Shortest,Cartesian, Index: 0"
    [8]: "Zipped, indecies: 0, 1 - Shortest,Cartesian, Index: 0,Cartesian, Index: 0"
    [9]: "Zipped, indecies: 0, 1 - Shortest,Cartesian, Index: 0,Cartesian, Index: 0,Cartesian, Index: 0"
    [10]: "Cartesian, Index: 1,Cartesian, Index: 1"
    [11]: "Zipped, indecies: 0, 1 - Shortest,Cartesian, Index: 1"
    [12]: "Zipped, indecies: 0, 1 - Shortest,Zipped, indecies: 0, 1 - Shortest"
    [13]: "Zipped, indecies: 0, 1 - Shortest,Zipped, indecies: 0, 1 - Shortest,Cartesian, Index: 0"
    [14]: "Zipped, indecies: 0, 1 - Shortest,Zipped, indecies: 0, 1 - Shortest,Cartesian, Index: 0,Cartesian, Index: 0"
    [15]: "Cartesian, Index: 1,Cartesian, Index: 1,Cartesian, Index: 1"
    [16]: "Zipped, indecies: 0, 1 - Shortest,Cartesian, Index: 1,Cartesian, Index: 1"
    [17]: "Zipped, indecies: 0, 1 - Shortest,Zipped, indecies: 0, 1 - Shortest,Cartesian, Index: 1"
    [18]: "Zipped, indecies: 0, 1 - Shortest,Zipped, indecies: 0, 1 - Shortest,Zipped, indecies: 0, 1 - Shortest"
    [19]: "Zipped, indecies: 0, 1 - Shortest,Zipped, indecies: 0, 1 - Shortest,Zipped, indecies: 0, 1 - Shortest,Cartesian, Index: 0"
    [20]: "Cartesian, Index: 1,Cartesian, Index: 1,Cartesian, Index: 1,Cartesian, Index: 1"
    [21]: "Zipped, indecies: 0, 1 - Shortest,Cartesian, Index: 1,Cartesian, Index: 1,Cartesian, Index: 1"
    [22]: "Zipped, indecies: 0, 1 - Shortest,Zipped, indecies: 0, 1 - Shortest,Cartesian, Index: 1,Cartesian, Index: 1"
    [23]: "Zipped, indecies: 0, 1 - Shortest,Zipped, indecies: 0, 1 - Shortest,Zipped, indecies: 0, 1 - Shortest,Cartesian, Index: 1"
    [24]: "Zipped, indecies: 0, 1 - Shortest,Zipped, indecies: 0, 1 - Shortest,Zipped, indecies: 0, 1 - Shortest,Zipped, indecies: 0, 1 - Shortest"
```

and it took visual studio approx 2 minutes ... though not sure if this was a visual studio issue, this seems bad. Let's avoid the toString and compare based on the properties.

Reply:
Yes, I have modified this to use the equals method in ReplicationInstructions class. Not using ToString() anymore.